x/crosschain/keeper/abci_test.go

```diff
@@ -449,3 +605,125 @@ func TestCheckAndUpdateCctxGasPrice(t *testing.T) {
 		})
 	}
 }
+
+func Test_CheckAndUpdateCCTXGasPriceBTC(t *testing.T) {
+	sampleTimestamp := time.Now()
+	gasRateUpdateInterval := observertypes.DefaultGasPriceIncreaseFlags.RetryInterval
+	retryIntervalReached := sampleTimestamp.Add(gasRateUpdateInterval + time.Second)
+
+	tt := []struct {
+		name           string
+		cctx           types.CrossChainTx
+		blockTimestamp time.Time
+		medianGasPrice uint64
+	}{
+		{
+			name: "can update fee rate",
+			cctx: types.CrossChainTx{
+				Index: "a",
+				CctxStatus: &types.Status{
+					CreatedTimestamp:    sampleTimestamp.Unix(),
+					LastUpdateTimestamp: sampleTimestamp.Unix(),
+				},
+				OutboundParams: []*types.OutboundParams{
+					{
+						ReceiverChainId: 8332,
+						CallOptions: &types.CallOptions{
+							GasLimit: 254,
+						},
+						GasPrice: "10",
+					},
+				},
+			},
+			blockTimestamp: retryIntervalReached,
+			medianGasPrice: 12,
+		},
+	}
+	for _, tc := range tt {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			k, ctx := testkeeper.CrosschainKeeperAllMocks(t)
+
+			// set block timestamp
+			ctx = ctx.WithBlockTime(tc.blockTimestamp)
+
+			// check and update gas rate
+			gasPriceIncrease, feesPaid, err := keeper.CheckAndUpdateCCTXGasPriceBTC(
+				ctx,
+				*k,
+				math.NewUint(tc.medianGasPrice),
+				tc.cctx,
+			)
+			require.NoError(t, err)
+
+			// check values
+			require.True(t, gasPriceIncrease.IsZero())
+			require.True(t, feesPaid.IsZero())
+
+			// check cctx if fee rate is updated
+			cctx, found := k.GetCrossChainTx(ctx, tc.cctx.Index)
+			require.True(t, found)
+			newGasPrice, err := cctx.GetCurrentOutboundParam().GetGasPriorityFeeUInt64()
+			require.NoError(t, err)
+			require.Equal(t, tc.medianGasPrice, newGasPrice)
+			require.EqualValues(t, tc.blockTimestamp.Unix(), cctx.CctxStatus.LastUpdateTimestamp)
+		})
+	}
+}
+
+func Test_IsCCTXGasPriceUpdateSupported(t *testing.T) {
```

Comment:
Can we add testNet chains to this unit test as well.

Reply:
added testnets https://github.com/zeta-chain/node/pull/3377/commits/279335bf423ac338cf96c984e6bb27df7e68eb61